src/org/labkey/test/components/ui/notifications/ServerNotificationMenu.java

```diff
@@ -0,0 +1,266 @@
+{
+
+    protected ServerNotificationMenu(WebElement element, WebDriver driver)
+    {
+        super(element, driver);
+    }
+
+    public static SimpleWebDriverComponentFinder<ServerNotificationMenu> finder(WebDriver driver)
+    {
+        return new MultiMenu.MultiMenuFinder(driver).withButtonId("server-notifications-button").wrap(ServerNotificationMenu::new);
+    }
+
+    /**
+     * Get the number of unread notifications. If no number is shown the implication is that there are no (0) unread notifications.
+     *
+     * @return The number of unread notifications.
+     */
+    public int getUnreadCount()
+    {
+        String text = "0";
+        boolean stale = true;
+
+        // Bit of a challenge to get the count. The element can update (because, you know it's async) if it does then a
+        // StaleElementException will happen. Try to protect against that by getting the element again f stale.
+        // If it is not there at all it will return null and exit the loop gracefully.
+        while(stale)
+        {
+            WebElement badge = Locator.tagWithClass("span", "badge").findElementOrNull(this);
+
+            if (badge != null)
+            {
+                try
+                {
+                    text = badge.getText().trim();
+                    stale = false;
+                }
+                catch (StaleElementReferenceException exception)
+                {
+                    // If the element went stale it may have been updated.
+                    stale = true;
+                }
+            }
+            else
+            {
+                stale = false;
+            }
+        }
+
+        return Integer.parseInt(text);
+
+    }
+
+    /**
+     * Check to see if a loading indicator, a spinner, is present.
+     *
+     * @return True if loading, false otherwise.
+     */
+    public boolean isLoading()
+    {
+        return getStatus().equals(MenuStatus.RUNNING);
+    }
+
+    /**
+     * Get the status as indicated by the icon shown.
+     *
+     * @return A {@link MenuStatus} enum.
+     */
+    public MenuStatus getStatus()
+    {
+        String status = elementCache().statusIcon().getAttribute("class").toLowerCase();
+
+        if(status.contains("fa-bell"))
+        {
+            return MenuStatus.COMPLETE;
+        }
+        else
+        {
+            return MenuStatus.RUNNING;
+        }
+    }
+
+    /**
+     * Is the 'Mark all as read' link visible?
+     *
+     * @return True if visible, false otherwise.
+     */
+    public boolean isMarkAllVisible()
+    {
+        return elementCache().markAll().isDisplayed();
+    }
+
+    /**
+     * Click the 'Mark all as read' link. This will cause the list to refresh, any references you have to a
+     * {@link ServerNotificationItem} will need to be reacquired.
+     */
+    public void clickMarkAll()
+    {
+        if(!isExpanded())
+        {
+            expand();
+        }
```

Comment:
This `if` is redundant. `expand()` checks whether the menu is expanded.
```suggestion
        expand();
```

Reply:
Changed.